scripts/check_missing_translations.py

```diff
@@ -0,0 +1,106 @@
+#! /usr/bin/python3
+
+import polib
+import git
+import sys
+import os
+import tempfile
+from collections import OrderedDict
+
+
+missing_keys = {}
+
+
+def list_entries(file):
+    entries = OrderedDict()
+
+    for entry in file:
+        if not entry.obsolete:
+            msgctxt = entry.msgctxt
+            if msgctxt is None:
+                msgctxt = ''
+
+            entries[(msgctxt, entry.msgid)] = entry
+
+    return entries
+
+def process_file(actual_file_path, previous_version_file_path, restore_missing):
+    actual_file = polib.pofile(actual_file_path, wrapwidth=10000)
+    previous_file = polib.pofile(previous_version_file_path, wrapwidth=10000)
+
+    previous_entries = list_entries(previous_file)
+    actual_entries = list_entries(actual_file)
+    changed = False
+
+    for key, entry in previous_entries.items():
+        if key not in actual_entries:
+            if key in missing_keys:
+                missing_keys[key].append(actual_file_path)
+            else:
+                missing_keys[key] = [actual_file_path]
+
+            if restore_missing:
+                # Find the entry that was just before the one we need to re-insert
+                previous_entry = None
+                for entry_in_previous_file in previous_file:
+                    if entry_in_previous_file == entry:
+                        break
+                    else:
+                        previous_entry = entry_in_previous_file
+
+                if previous_entry is None:
+                    # The entry was at the very beginning
+                    actual_file.insert(0, entry)
+                else:
+                    # Now find the position in the new file and re-insert the missing entry
+                    index = 0
+                    inserted = False
+                    for entry_in_actual_file in actual_file:
+                        if entry_in_actual_file.msgctxt == previous_entry.msgctxt and entry_in_actual_file.msgid == previous_entry.msgid:
+                            actual_file.insert(index + 1, entry)
+                            inserted = True
+                            break
+                        else:
+                            index += 1
+
+                    if not inserted:
+                        actual_file.append(entry)
+
+                changed = True
+
+    if changed:
+        print(f"Fixed file {actual_file_path}")
+        actual_file.save()
+
+
+try:
+    previous_version = sys.argv[1]
+    restore_missing = "--restore-missing" in sys.argv
+except:
+    print("Compares the translations present in the current folder with the same files from an other commit/branch. This will write a report.csv file containing the missing translations and the locations where they were expected. Make sure to run this script at the root of Cura/Uranium")
```

Comment:
can use argparser maybe to make it a bit clearer?

Reply:
Well I have very few arguments so I didn't want to go there, but that would make the script clearer indeed.

Reply:
Done, and this makes the script much more user friendly indeed :+1: